### PR TITLE
Adding more sort methods. Fixes #36

### LINKED ISSUE #36: Update docs for sort methods
Some of the new ones are missing.

https://join.lemmy.ml/docs/en/about/guide.html#sorting

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

File: src/en/about/guide.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ Type | Description
 Active | Trending sort based on the score, and the most recent comment time.
 Hot | Trending sort based on the score, and the post creation time.
 New | Newest items.
+Most Comments | The posts with the most comments.
+New Comments | The posts that have new comments, IE a forum-style sort.
```
Comment (Nutomic) [inline src/en/about/guide.md:19]: "The posts that have most recently been commented on". Might be a bit clearer, but I'm not sure how to explain it.

Comment (dessalines) [reply src/en/about/guide.md:19]: I'll say, "The posts with recent comments"

Comment (Nutomic) [reply src/en/about/guide.md]: "The post with *most* recent comments" sounds clearer to me.

Sorry for nitpicking so much haha

Comment (dessalines) [reply src/en/about/guide.md]: Haha that's okay. updated

**Pull request merged**